contracts/interfaces/workflows/IRoyaltyWorkflows.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.26;
+
+/// @title Royalty Workflows Interface
+/// @notice Interface for IP royalty workflows.
+interface IRoyaltyWorkflows {
+    /// @notice Details for claiming royalties from a child IP.
+    /// @param childIpId The address of the child IP.
+    /// @param royaltyPolicy The address of the royalty policy.
+    /// @param currencyToken The address of the currency (revenue) token to claim.
+    /// @param amount The amount of currency (revenue) token to claim.
+    struct RoyaltyClaimDetails {
+        address childIpId;
+        address royaltyPolicy;
+        address currencyToken;
+        uint256 amount;
+    }
+
+    /// @notice Transfers royalties from royalty policy to the ancestor IP's royalty vault, takes a snapshot,
+    /// and claims revenue on that snapshot for each specified currency token.
+    /// @param ancestorIpId The address of the ancestor IP.
+    /// @param claimer The address of the claimer of the revenue tokens (must be a royalty token holder).
+    /// @param currencyTokens The addresses of the currency (revenue) tokens to claim (each address must be unique).
+    /// @param royaltyClaimDetails The details of the royalty claim from child IPs,
+    /// see {IRoyaltyWorkflows-RoyaltyClaimDetails}.
+    /// @return snapshotId The ID of the snapshot taken.
+    /// @return amountsClaimed The amount of revenue claimed for each currency token.
+    function transferToVaultAndSnapshotAndClaimByTokenBatch(
+        address ancestorIpId,
+        address claimer,
+        address[] calldata currencyTokens,
+        RoyaltyClaimDetails[] calldata royaltyClaimDetails
+    ) external returns (uint256 snapshotId, uint256[] memory amountsClaimed);
```

Comment:
Why do we need pass an array of currencyTokens while `RoyaltyClaimDetails` already has the currentcyToken info?

Reply:
The `claimByTokenBatch` function requires the `currencyTokens` array to be unique. If the same currency token is passed more than once, it will trigger a `IpRoyaltyVault__NoClaimableTokens` error when attempting to claim it again. However, `royaltyClaimDetails` may include the same currency token multiple times because different descendant IPs can be licensed under the same currency token. 

We could remove the `currencyTokens` param, we would then need to filter out duplicated currency tokens from `royaltyClaimDetails` within the contract.

Reply:
I see. 
Do we expect all `royaltyClaimDetails.currencyToken` should be included in the unique array `currencyTokens`?
If so, we need to add check for it and revert otherwise, or we our function may execute successfully with unexpected result.
In that case, seems extract a unique `crurrencyTokens` from array of `royaltyClaimDetails` is simpler. 
I think we can merge the PR as is, and using another PR address this comment.

Reply:
Yes, agreed. Will have another PR to extract `currencyTokens` from `royaltyClaimDetails`.

Reply:
Resolved in #75